bftengine/src/bftengine/KeyManager.cpp

```diff
@@ -294,12 +297,32 @@ bool KeyManager::KeysView::load() {
   return true;
 }
 
-void KeyManager::PersistentSaverLoader::save(const std::string& str) {
-  std::string nonConstStr = str;
-  LOG_DEBUG(KEY_EX_LOG, "Persistent save size " << nonConstStr.size());
-  ps->setKeysView(nonConstStr);
+void KeyManager::FileSecureStore::save(const std::string& str) {
+  if (str.empty()) {
+    LOG_INFO(KEY_EX_LOG, "Got empty string to save to key file");
+    return;
+  }
+  std::ofstream myfile;
+  myfile.open(fileName.c_str());
+  if (!myfile.good()) {
+    LOG_FATAL(KEY_EX_LOG, "Couldn't save key file to " << fileName);
+    ConcordAssert(false);
+  }
+  myfile << str;
+  myfile.close();
 }
 
-std::string KeyManager::PersistentSaverLoader::load() { return ps->getKeysView(); }
+std::string KeyManager::FileSecureStore::load() {
+  std::ifstream inFile;
+  inFile.open(fileName.c_str());
+  if (!inFile.good()) {
+    LOG_WARN(KEY_EX_LOG, "key file wasn't loaded " << fileName);
+    return "";
```

Comment:
Nit: If the empty string is to identify a failure, it might be better to change the return type to `std::optional<std::string>>`.

Reply:
It's a good idea, I'll refactor it in a next PR